### PR TITLE
preview image before update

### PR DESCRIPTION
**Pull request by VuongLong**

### CODE REVIEW

File: app/helpers/users_helper.rb [hunk 1]
```diff
@@ -1,9 +1,15 @@
 module UsersHelper
+
```
Comment (ThuBM) [inline app/helpers/users_helper.rb]: remove line

File: app/helpers/users_helper.rb [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
-    gravatar_url = "https://secure.gravatar.com/avatar/#{gravatar_id}?s=#{size}"
-    image_tag gravatar_url, alt: user.name, class: "gravatar"
+    gravatar_url = if user.avatar? 
+      user.picture.url
+    else 
+      "https://secure.gravatar.com/avatar/#{gravatar_id}?s=#{size}" 
+    end
+    image_tag gravatar_url, alt: user.name, class: "gravatar",
+      size: "#{size}x#{size}", id: "user_image"
   end
```
Comment (ThuBM) [inline app/helpers/users_helper.rb]: remove line

File: app/helpers/users_helper.rb [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
+  
   def gravatar_for user, options = {size: Settings.user.gravatar.image_size}
     gravatar_id = Digest::MD5::hexdigest user.email.downcase
     size = options[:size]
-    gravatar_url = "https://secure.gravatar.com/avatar/#{gravatar_id}?s=#{size}"
-    image_tag gravatar_url, alt: user.name, class: "gravatar"
+    gravatar_url = if user.avatar? 
+      user.picture.url
+    else 
+      "https://secure.gravatar.com/avatar/#{gravatar_id}?s=#{size}"
```
Comment (ThuBM) [inline app/helpers/users_helper.rb]: Settings

File: app/models/user_course.rb [hunk 4]
```diff
@@ -3,6 +3,7 @@ class UserCourse < ActiveRecord::Base
   belongs_to :course
   has_many :user_course_subjects, dependent: :destroy
   has_many :subjects, through: :user_course_subjects
-  
-  scope :active_course, ->{where "status = ?", Settings.status.active}
+
+  scope :active_user_course, ->user_id, course_id{where "user_id = ?
```
Comment (ThuBM) [inline app/models/user_course.rb]: active_user_course không so sánh status = active à???

**Comment by ThuBM** [comment]
OK

**Comment by thiennv57** [comment]
OK

**Comment by binhthanhnguyen** [comment]
okay

File: app/assets/javascripts/preview_image_before_upload.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+function PreviewImage(image_id) {
+  var oFReader = new FileReader();
+  oFReader.readAsDataURL(document.getElementById("user_picture").files[0]);
+
+  oFReader.onload = function(oFREvent) {
+    document.getElementById(image_id).src = oFREvent.target.result;
```
Comment (chuanhtuan) [inline app/assets/javascripts/preview_image_before_upload.js]: jquery, plz

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -3,8 +3,10 @@ class UsersController < ApplicationController
   before_action :authenticate_user!
 
   def show
-    @user_course = @user.user_courses.active_course.first
-    @course = @user_course.course
+    @course = @user.courses.active_course.first
+    unless @course.nil? 
+      @user_course = UserCourse.active_user_course(@user.id, @course.id).first
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: user.user_courses.find_by course_id: abcxyz

File: app/models/user_course.rb [hunk 7]
```diff
@@ -4,5 +4,6 @@ class UserCourse < ActiveRecord::Base
   has_many :user_course_subjects, dependent: :destroy
   has_many :subjects, through: :user_course_subjects
 
-  scope :active_course, ->{where "status = ?", Settings.status.active}
+  scope :active_user_course, ->user_id, course_id{where "user_id = ?
```
Comment (chuanhtuan) [inline app/models/user_course.rb]: remove

**Pull request merged**